src/context/yaml/handlers/clientGrants.ts

```diff
@@ -11,13 +12,34 @@ async function parse(context: YAMLContext): Promise<ParsedClientGrants> {
 
   if (!clientGrants) return { clientGrants: null };
 
-  return {
-    clientGrants,
-  };
+  const shouldPreserveKeywords =
+    // @ts-ignore because the string=>boolean conversion may not have happened if passed-in as env var
+    context.config.AUTH0_PRESERVE_KEYWORDS === 'true' ||
+    context.config.AUTH0_PRESERVE_KEYWORDS === true;
+
+  if (shouldPreserveKeywords) {
+    // AUTH0_PRESERVE_KEYWORDS need client id instead for client name
+    // here can not use client from context.assets because it does not have the client_id
+    const clients = await context.mgmtClient.clients.getAll({
+      paginate: true,
+      include_totals: true,
+    });
+
+    return {
+      clientGrants: clientGrants.map((grant) => {
+        const dumpGrant = { ...grant };
+        dumpGrant.client_id = convertClientNameToId(dumpGrant.client_id, clients || []);
```

Comment:
Sorry, I didn't catch this right before 🤦‍♂️ .

We shouldn't be dumping the client ID because it isn't going to be the same on each tenant. Remember that the Deploy CLI is a multi-tenant configuration management tool. The only way to maintain agnostic to tenant is by continuing to use name over ID.